test/test_abort_blocks_after.py

```diff
@@ -0,0 +1,143 @@
+    assert_transaction(contract_deploy_info["tx_hash"], "REJECTED")
+
+    # Test RPC get block status mapping from ABORTED to REJECTED
+    rpc_aborted_block = rpc_call(
+        "starknet_getBlockWithTxs",
+        params={
+            "block_id": {"block_hash": rpc_felt(contract_deploy_block["block_hash"])}
+        },
+    )
+    assert rpc_aborted_block["result"]["status"] == "REJECTED"
+
+
+@devnet_in_background(*PREDEPLOY_ACCOUNT_CLI_ARGS)
+def test_abort_many_blocks_many_transactions():
+    """Test abort of many blocks and many transactions."""
+
+    # Block and transaction should be accepted on L2
+    contract_deploy_info = deploy(CONTRACT_PATH, inputs=["0"])
+    contract_deploy_block = get_block(parse=True)
+    assert contract_deploy_block["status"] == "ACCEPTED_ON_L2"
+    assert_tx_status(contract_deploy_info["tx_hash"], "ACCEPTED_ON_L2")
+
+    # Block and transaction should be accepted on L2
+    invoke_tx_hash = invoke(
+        calls=[(contract_deploy_info["address"], "increase_balance", [10, 20])],
+        account_address=PREDEPLOYED_ACCOUNT_ADDRESS,
+        private_key=PREDEPLOYED_ACCOUNT_PRIVATE_KEY,
+    )
+    invoke_block = get_block(parse=True)
+    assert invoke_block["status"] == "ACCEPTED_ON_L2"
+    assert_transaction(invoke_tx_hash, "ACCEPTED_ON_L2")
+
+    # Blocks should be aborted and transactions should be rejected
+    response = abort_blocks(contract_deploy_block["block_hash"])
+    assert response.status_code == 200
+    assert response.json()["aborted"] == [
+        contract_deploy_block["block_hash"],
+        invoke_block["block_hash"],
+    ]
+    last_block = get_block(parse=True)
+    assert last_block["status"] == "ACCEPTED_ON_L2"
+    assert last_block["block_number"] == 0
+    contract_deploy_block_after_abort = get_block(
+        block_hash=contract_deploy_block["block_hash"], parse=True
+    )
+    assert contract_deploy_block_after_abort["status"] == "ABORTED"
+    assert_transaction(contract_deploy_info["tx_hash"], "REJECTED")
+    invoke_block_after_abort = get_block(
+        block_hash=invoke_block["block_hash"], parse=True
+    )
+    assert invoke_block_after_abort["status"] == "ABORTED"
+    assert_transaction(invoke_tx_hash, "REJECTED")
+
+    # Test RPC get block status mapping from ABORTED to REJECTED
+    rpc_aborted_block_contract = rpc_call(
+        "starknet_getBlockWithTxs",
+        params={
+            "block_id": {"block_hash": rpc_felt(contract_deploy_block["block_hash"])}
+        },
+    )
+    assert rpc_aborted_block_contract["result"]["status"] == "REJECTED"
+    rpc_aborted_block_invoke = rpc_call(
+        "starknet_getBlockWithTxs",
+        params={"block_id": {"block_hash": rpc_felt(invoke_block["block_hash"])}},
+    )
+    assert rpc_aborted_block_invoke["result"]["status"] == "REJECTED"
+
+    # Block and transaction should be accepted on L2
+    contract_deploy_info = deploy(CONTRACT_PATH, inputs=["0"])
```

Comment:
Fixed in 3965c31efc77337cb98c64357c4d5a28a85b9c4c

Reply:
I feel like there is a lot of similarity between this test and the previous couple of tests. What I meant is that there could be a test dedicated only to generating new blocks after aborting. So like: create a block, abort it, generate a new block, check that the new block is correct (without making other assertions since we already have them in other tests).

Reply:
True, fixed in 212db97a85408350386080013997291f10e18264